Source/Core/Duality/Utility/Coroutines/WaitUntil.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// Struct in charge of managing the timings of a Coroutine's execution
+	/// </summary>
+	public struct WaitUntil
+	{
+		private enum WaitType
+		{
+			Invalid,
```

Comment:
I would remove the Invalid state and just let it default to Frames. It doesn't really add anything else than yet another if to check for this state.  If it defaults to Frames and a internal value of 0 it will just act like a condition that completes instantly.